Sources/Orbit/Support/Toast/ToastQueue.swift

```diff
@@ -0,0 +1,160 @@
+    }
+    
+    /// View model for Orbit Toast component.
+    public struct Toast: Identifiable {
+        public let id: UUID
+        let description: String
+        let icon: Icon.Symbol
+        let progress: Double
+
+        init(id: UUID = UUID(), description: String, icon: Icon.Symbol, progress: Double) {
+            self.id = id
+            self.description = description
+            self.icon = icon
+            self.progress = progress
+        }
+
+        public init(_ description: String, icon: Icon.Symbol = .none) {
+            self.init(description: description, icon: icon, progress: 0)
+        }
+
+        func withProgress(_ progress: Double) -> Self {
+            .init(id: id, description: description, icon: icon, progress: min(1, progress))
+        }
+    }
+
+    private var cancellable: AnyCancellable?
+    private var toastsSubject = PassthroughSubject<Toast, Never>()
+    private var currentToastActionSubject: CurrentValueSubject<ToastAction, Never>?
+    
+    /// Currently active Toast.
+    @Published private(set) var toast: Toast?
+
+    public init() {
+        cancellable = toastsSubject
+            .buffer(size: Self.toastsBufferSize, prefetch: .keepFull, whenFull: .dropOldest)
+            .delay(for: .seconds(Self.appearDelay), scheduler: RunLoop.main)
+            .flatMap(maxPublishers: .max(1)) { [weak self] toast in
+                self?.toastPublisher(for: toast) ?? Just(nil).eraseToAnyPublisher()
+            }
+            .sink { [weak self] toast in
+                self?.processToast(toast)
+            }
+    }
+
+    /// Add a new Toast to be displayed as soon as there is no active Toast displayed.
+    public func add(_ desctription: String, icon: Icon.Symbol = .none) {
+        add(Toast(desctription, icon: icon))
+    }
+
+    /// Add a new Toast to be displayed as soon as there is no active Toast displayed.
+    public func add(_ toast: Toast) {
+        toastsSubject.send(toast)
+    }
+    
+    /// Pause the active Toast dismiss progress.
+    public func pause() {
+        currentToastActionSubject?.send(.pause)
+    }
+    
+    /// Resume the active Toast dismiss progress.
+    public func resume() {
+        currentToastActionSubject?.send(.run)
+    }
+    
+    /// Dismiss currently active Toast.
+    public func dismiss() {
+        currentToastActionSubject?.send(.dismiss)
+    }
+    
+    private func toastPublisher(for toast: Toast) -> AnyPublisher<Toast?, Never> {
+        let currentToastActionSubject = CurrentValueSubject<ToastAction, Never>(.run)
+        self.currentToastActionSubject = currentToastActionSubject
+     
+        var timerPublisher: AnyPublisher<Void, Never> {
+            Timer
+                .publish(every: 0.1, on: RunLoop.main, in: .common)
+                .autoconnect()
+                .map { _ in Void() }
+                .eraseToAnyPublisher()
+        }
+        
+        func incrementFor(action: ToastAction) -> TimeInterval {
```

Comment:
I'd name it either `increment(for action:` or `increment(forAction:`. Internal code style suggests the first one since the type is quite specialized.

Reply:
Used the second one, because the "action" would be lost in the combine chain (and extracting to method was mostly for readability)